test/lib/sstable_utils.hh

```diff
@@ -245,6 +247,65 @@ future<> for_each_sstable_version(AsyncAction action) {
     return seastar::do_for_each(all_sstable_versions, std::move(action));
 }
 
+class sst_factory {
+public:
+    test_env& env;
+    schema_ptr schema;
+    tmpdir& tmp;
+    sstables::sstable_version_types version = sstables::get_highest_sstable_version();
+    size_t buffer_size = sstables::default_sstable_buffer_size;
+private:
+    generation_factory local_gen_factory;
+    generation_factory& gen_factory = local_gen_factory;
```

Comment:
Note, that test_env::make_sstable() can already produce sstables with "monotonic" generations itself. It has the generation counter on board since be7f4ff53a73a7a0817629160a5ac7e306a1b616 . So this factory is very likely duplicating this functionality.

I'm not proposing to refactor it here though

Reply:
yeah, when I originally wrote ths code code be7f4ff53a73a7a0817629160a5ac7e306a1b616 was either in your branch or not born yet incepted